api/v1alpha1/hybrisapp_types.go

```diff
@@ -41,6 +41,15 @@ type HybrisAppSpec struct {
 
 	// Hybris app repository source location
 	SourceRepoContext string `json:"sourceRepoContext,omitempty"`
+
+	// Hybris app repository local.properties override location
+	SourceRepoLocalPorpertiesOverride string `json:"sourceRepoLocalPropertiesOverride,omitempty"`
```

Comment:
For optional spec, it is better to use pointer and check if it is `nil` in the controller before using it.
```suggestion
	SourceRepoLocalPorpertiesOverride *string `json:"sourceRepoLocalPropertiesOverride,omitempty"`
```

Reply:
I do not see these types being different from the existing ones, that the existing design needs to be altered. There is a check happening in assemble script, and the status gets ignored if the values are empty.

Reply:
The existing ones are not optional. If changing them to use pointers, the pointers will never be `nil`.

Reply:
I see all the existing ones being optional (omitempty) and being capable of being set with a nill value. Why do you claim they cannot be nil? I don't want to invalidate my testing now, by changing it to the pointer at this time. I will do that if future changes come up

	BaseImageName string `json:"baseImageName,omitempty"`

	// Hybris base image tag
	// +kubebuilder:validation:Required
	BaseImageTag string `json:"baseImageTag,omitempty"`

	// Hybris app source repository URL
	// +kubebuilder:validation:Required
	SourceRepoURL string `json:"sourceRepoURL,omitempty"`

	// Hybris app source repository reference
	// +kubebuilder:validation:Required
	SourceRepoRef string `json:"sourceRepoRef,omitempty"`

	// Hybris app repository source location
	SourceRepoContext string `json:"sourceRepoContext,omitempty"`

	// Hybris app repository local.properties override location
	SourceRepoLocalPorpertiesOverride string `json:"sourceRepoLocalPropertiesOverride,omitempty"`

	// Hybris app Apache server.xml jvmRoute name
	ApachejvmRouteName string `json:"apachejvmRouteName,omitempty"`

	// Hybris app ANT tasks
	HybrisANTTaskNames string `json:"hybrisANTTaskNames,omitempty"`

Reply:
Because of this annotation `kubebuilder:validation:Required`.

Reply:
I don't see the code-breaking if these new properties are omitted. I tested now with a HybriApp CR having only the below properties without the new ones included. It behaved just fine.

HybrisApp CR----->
  spec:
    baseImageName: hybris-base
    baseImageTag: v0.2
    sourceRepoRef: master
    sourceRepoURL: https://github.com/faizalak/sap-commerce-app-example.git
    
Build---->
STEP 7: USER 1001
STEP 8: RUN /usr/local/s2i/assemble
---> Copying Hybris application source cp -r /tmp/src/hybris/* /opt/hybris
---> Hybris config /opt/hybris/config/local.properties present in the source repo will be used as is
---> Apache config /opt/hybris/config/tomcat/conf/server.xml present in the source repo will be used as is
---> Building SAP Commerce ...
---> No Hybris ANT tasks to execute
---> Executing >ant with default task clean all
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=400m; support was removed in 8.0
Buildfile: /opt/hybris/bin/platform/build.xml